aten/src/ATen/native/mps/operations/ReduceOps.mm

```diff
@@ -139,6 +139,8 @@ void reduction_out_mps(
   MPSReductionType reduction_type,
   const std::string& func_name) {
 
+  // issue 103641234, reduction ops does not have int64 support
+  TORCH_WARN_ONCE(input_t.scalar_type() != ScalarType::Long, "MPS: no support for int64 reduction ops, casting it to int32");
```

Comment:
Err, wouldn't it do the opposite? (I.e. print the warning for all types but int64?) 
For example see:
```
% python -c "import torch;x=torch.arange(1., 10.,device='mps').view(3, 3).trace()"
<string>:1: UserWarning: 1MPS: no support for int64 reduction ops, casting it to int32 (Triggered internally at /Users/malfet/git/pytorch/pytorch/aten/src/ATen/native/mps/operations/ReduceOps.mm:143.)
```
Will fix it in the same PR that introduces fail

Reply:
Actually, let me do it in separate PR, as this is not how it supposed to be done at all.
Here is the PR https://github.com/pytorch/pytorch/pull/95559